x-pack/libbeat/reader/decoder/config.go

```diff
@@ -11,25 +11,38 @@ import (
 )
 
 // decoderConfig contains the configuration options for instantiating a decoder.
-type decoderConfig struct {
-	Codec *codecConfig `config:"codec"`
+type DecoderConfig struct {
```

Comment:
This could lose its stutter.
```suggestion
type Config struct {
```